### PR TITLE
[test] Adapt MCH checks to CrayPE 17.06 on Kesch (RH7.5)

### PR DESCRIPTION
**Pull request by lucamar**
I have tested the MCH checks using the Cray Programming Environment 17.06 (module `PE/17.06`) and the MPI library `openmpi/4.0.1-pgi-18.5-gcc-5.4.0-2.26-cuda-8.0.61` for `PrgEnv-pgi/18.5`. 
The latter is not yet in production, since is part of the pull request https://github.com/eth-cscs/production/pull/1271, therefore I had to use my sandbox for testing it.
Fieldextra checks cannot work at present since the module `fieldextra/12.7.5-gmvolf-17.02` is not yet in production.

### CODE REVIEW

**Comment by lucamar** [comment]
The module `OpenMPI` is now available in `PrgEnv-pgi`, therefore the checks will run in production.

**Comment by teojgo** [comment]
@jenkins-cscs retry kesch

**Comment by lucamar** [comment]
I have updated the reference value of `CudaStressTest`, since it was failing occasionally. 
The change is from 2.12769 to 2.25, an increase of ~ 5.7%.

**Comment by lucamar** [comment]
The failure of the checks `HaloExchangeTest_nocomp` and `HaloExchangeTest_default` on 144 MPI tasks (i.e. 9 nodes) is unusual...

**Comment by vkarak** [comment]
@lucamar I see very bad performance for the `HaloExchangeTest_default` test. Is this expected?

```
Reason: performance error: failed to meet reference: elapsed_time=30.3839, expected 5.53777 (l=-inf, u=6.3684354999999995)
```

File: cscs-checks/mch/gpu_direct_cuda.py [hunk 1]
```diff
@@ -22,7 +22,6 @@ def __init__(self):
         elif self.current_system.name == 'kesch':
             self.exclusive_access = True
             self.valid_prog_environs = ['PrgEnv-gnu']
-            self.modules = ['craype-accel-nvidia35']
```
Comment (vkarak) [inline cscs-checks/mch/gpu_direct_cuda.py]: Not even `cudatoolkit` is needed here?

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 2]
```diff
@@ -47,6 +67,18 @@ def __init__(self):
     def setup(self, partition, environ, **job_opts):
         if environ.name.startswith('PrgEnv-cray'):
             self.build_system.fflags += ['-hacc', '-hnoomp']
+
+        elif environ.name.startswith('PrgEnv-cce'):
+            self.build_system.fflags += ['-hacc', '-hnoomp']
+            if self.current_system.name == 'arolla':
+                self.build_system.ldflags = [
+                    '-L/cm/shared/apps/cuda92/toolkit/9.2.88/lib64',
+                    '-lcublas', '-lcudart']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 3]
```diff
@@ -47,6 +67,18 @@ def __init__(self):
...
+            elif self.current_system.name == 'tsa':
+                self.build_system.ldflags = [
+                    '-L/cm/shared/apps/cuda10.0/toolkit/10.0.130/lib64',
+                    '-lcublas', '-lcudart']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 4]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
+                self.build_system.ldflags = [
+                    '-acc', '-ta:tesla:cc70,cuda10.0', '-lstdc++',
+                    '-L/cm/shared/apps/cuda92/toolkit/9.2.88/lib64',
+                    '-lcublas', '-lcudart']
+            elif self.current_system.name == 'tsa':
+                self.build_system.fflags += ['-ta=tesla,cc70,cuda10.0']
+                self.build_system.ldflags = [
+                    '-acc', '-ta:tesla:cc70,cuda10.0', '-lstdc++',
+                    '-L/cm/shared/apps/cuda10.0/toolkit/10.0.130/lib64',
+                    '-lcublas', '-lcudart']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 5]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
+                self.build_system.ldflags = [
+                    '-acc', '-ta:tesla:cc35,cuda8.0', '-lstdc++',
+                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
+                    '-lcublas', '-lcudart']
+            elif self.current_system.name == 'arolla':
+                self.build_system.fflags += ['-ta=tesla,cc70,cuda10.0']
+                self.build_system.ldflags = [
+                    '-acc', '-ta:tesla:cc70,cuda10.0', '-lstdc++',
+                    '-L/cm/shared/apps/cuda92/toolkit/9.2.88/lib64',
+                    '-lcublas', '-lcudart']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 6]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
-                                             '-lstdc++']
-                if environ.name == 'PrgEnv-pgi-nompi':
-                    self.build_system.ldflags += [
-                        '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
-                        '-lcublas', '-lcudart'
-                    ]
+                self.build_system.ldflags = [
+                    '-acc', '-ta:tesla:cc35,cuda8.0', '-lstdc++',
+                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
+                    '-lcublas', '-lcudart']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 7]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
             self.build_system.ldflags = ['-lstdc++']
             if self.current_system.name == 'kesch':
                 self.build_system.ldflags += [
-                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
-                    '-lcublas', '-lcudart'
-                ]
+                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64']
+            if self.current_system.name == 'arolla':
+                self.build_system.ldflags += [
+                    '-L/cm/shared/apps/cuda92/toolkit/9.2.88/lib64']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 8]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
+                    '-lcublas', '-lcudart']
+
         elif environ.name.startswith('PrgEnv-gnu'):
             self.build_system.ldflags = ['-lstdc++']
             if self.current_system.name == 'kesch':
                 self.build_system.ldflags += [
-                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
-                    '-lcublas', '-lcudart'
-                ]
+                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 9]
```diff
@@ -55,19 +87,34 @@ def setup(self, partition, environ, **job_opts):
...
+            if self.current_system.name == 'arolla':
+                self.build_system.ldflags += [
+                    '-L/cm/shared/apps/cuda92/toolkit/9.2.88/lib64']
+            if self.current_system.name == 'tsa':
+                self.build_system.ldflags += [
+                    '-L/cm/shared/apps/cuda10.0/toolkit/10.0.130/lib64']
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Please move the closing brackets to the next line.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 10]
```diff
@@ -17,23 +19,41 @@ def __init__(self):
...
+            self.modules = ['cudatoolkit/8.0.61']
             self.num_tasks = 8
             self.num_tasks_per_node = 8
             self.num_gpus_per_node = 8
             self.build_system.options = ['NVCC_FLAGS="-arch=compute_37"']
-            # FIXME: temporary workaround until the mvapich module is fixed;
-            #        'PrgEnv-{pgi,gnu}-c2sm-gpu' will be added later
-            self.valid_prog_environs += ['PrgEnv-cray-c2sm-gpu']
-
-        if self.current_system.name in ['daint', 'dom']:
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Here you are removing the configuration for daint/dom, but these systems are in the `valid_systems`

**Comment by lucamar** [comment]
> @lucamar I see very bad performance for the `HaloExchangeTest_default` test. Is this expected?
> 
> ```
> Reason: performance error: failed to meet reference: elapsed_time=30.3839, expected 5.53777 (l=-inf, u=6.3684354999999995)
> ```

Unfortunately it was not happening so often in the past: we will need to monitor the check once we merge it.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py (hunk 10 above)
Comment (lucamar) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: The `if` statement for Dom and Piz Daint has been moved above, on line 17

**Pull request merged**